ffi/diplomat/src/provider.rs

```diff
@@ -73,10 +73,17 @@ pub mod ffi {
                 feature = "provider_fs",
                 not(any(target_arch = "wasm32", target_os = "none"))
             ))]
-            icu_provider_fs::FsDataProvider::try_new(path)
-                .map_err(Into::into)
-                .map(convert_buffer_provider)
-                .into()
+            {
+                // #2520
+                // In the future we can start using OsString APIs to support non-utf8 paths
+                if core::str::from_utf8(path.as_bytes()).is_err() {
+                    return Err(ICU4XError::DataIoError).into();
```

Comment:
Suggestion: Log the error here, since normally when creating an ICU4XError we also log what the original error was.